py/server/tests/test_keyed_record_adapter.py

```diff
@@ -0,0 +1,301 @@
+import unittest
+from typing import List
+
+import jpy
+
+from deephaven import new_table
+from deephaven.column import *
+from deephaven.constants import *
+from deephaven.experimental import *
+from tests.testbase import BaseTestCase
+
+_JTableTools = jpy.get_type("io.deephaven.engine.util.TableTools")
+
+
+class KraTestCase(BaseTestCase):
+
+    def testCustomKeyedRecordAdapterWithOneStrKeyCol(self):
+        # source = Table(_JTstTools.testRefreshingTable(
+        #     _JTstTools.i(2, 4).copy().toTracking(),
+        #     _JTableTools.intCol("Sym", 'AAPL', None),
+        #     _JTableTools.doubleCol("Price", 1.1, NULL_DOUBLE),
+        #     _JTableTools.longCol("Size", 10_000_000_000, NULL_LONG),
+        #     _JTableTools.col("Exch", "ARCA", None),
+        # ))
+
+        source = new_table([
+            string_col("Sym", ['AAPL', None]),
+            double_col("Price", [1.1, NULL_DOUBLE]),
+            long_col("Size", [10_000_000_000, NULL_LONG]),
+            string_col("Exch", ["ARCA", None]),
+        ])
+        _JTableTools.show(source.j_table)
+
+        class StockTrade:
+            sym: str
+            price: float
+            size: int
+            exch: str
+
+            def __init__(self, sym: str, price: float, size: int, exch: str):
+                self.sym = sym
+                self.price = price
+                self.size = size
+                self.exch = exch
+
+        keyed_record_adapter: KeyedRecordAdapter[str, StockTrade] = KeyedRecordAdapter(
```

Comment:
type hints are not normally included in a place like this, since they can be inferred.